docs: update supabase docs for nextjs 16 proxyts change

ChrisChinchilla opened the pull request:
## What kind of change does this PR introduce?

Docs update

## What is the current behaviour?

Nextjs examples to use v16 `proxy`

## What is the new behaviour?

Update docs to match same code as [Nextjs with Supabase Template](https://github.com/vercel/next.js/tree/canary/examples/with-supabase)

- closes #40851

ChrisChinchilla committed:
Quickstart next 16 update

ChrisChinchilla committed:
Fix paths and env vars

kallebysantos committed:
docs: refactor nextjs server-side auth to use Proxy instead of middleware

kallebysantos committed:
docs: refactor nextjs server-side auth to match proxy

kallebysantos committed:
docs: refactor nextjs example to match Proxy

kallebysantos committed:
docs: refactor nextjs auth AI prompt to match Proxy

kallebysantos committed:
docs: refactor nextjs sentry telemetry integration to match Proxy

kallebysantos committed:
examples: update nextjs realtime example to match middleware

kallebysantos committed:
docs: refactoring guides to use nextjs proxy

dijonmusters commented on apps/docs/content/guides/getting-started/quickstarts/nextjs.mdx:
Unrelated to the changes you have made here, but probably good to clean as we go - Step 4 can be removed from this guide as this file is already in the `with-supabase` template

ChrisChinchilla replied:
This was partially true… Turns out that we need to change the path though…

dijonmusters commented on apps/docs/content/guides/getting-started/quickstarts/nextjs.mdx:
````diff
@@ -53,7 +53,7 @@ hideToc: true
         ```text name=.env.local
         NEXT_PUBLIC_SUPABASE_URL=<SUBSTITUTE_SUPABASE_URL>
-        NEXT_PUBLIC_SUPABASE_PUBLISHABLE_KEY=<SUBSTITUTE_SUPABASE_PUBLISHABLE_KEY>
+        NEXT_PUBLIC_SUPABASE_PUBLISHABLE_DEFAULT_KEY=<SUBSTITUTE_SUPABASE_PUBLISHABLE_KEY>
````
Let's keep these consistent as `NEXT_PUBLIC_SUPABASE_PUBLISHABLE_KEY`

dijonmusters commented on apps/docs/content/guides/getting-started/quickstarts/nextjs.mdx:
```diff
@@ -76,34 +76,32 @@ hideToc: true
         return createServerClient(
           process.env.NEXT_PUBLIC_SUPABASE_URL!,
-          process.env.NEXT_PUBLIC_SUPABASE_PUBLISHABLE_KEY!,
+          process.env.NEXT_PUBLIC_SUPABASE_PUBLISHABLE_DEFAULT_KEY!,
```
Let's keep these consistent as `NEXT_PUBLIC_SUPABASE_PUBLISHABLE_KEY`

kallebysantos replied:
solved 4db07d2ec58183247864d015bcd73275f3ba4b32

dijonmusters reviewed (dismissed):
This looks great! Couple of small fixes. Thank you so much for updating this 🙌

kallebysantos committed:
examples: update nextjs-full example to match Next16 template

kallebysantos committed:
example: update nextjs-user-management to match nextjs 16

kallebysantos committed:
docs: refactoring nextjs user-management tutorial to use typescript only

kallebysantos committed:
docs: refactoring nextjs quickstart, removing step 4

since this step is already included on `with-supabase` template, we can
just remove this redundant step

kallebysantos committed:
docs: auth-helpers nextjs pages, Nextjs16 proxy disclaimer

kallebysantos committed:
stamp: lint

kallebysantos committed:
stamp: revert 'NEXT_PUBLIC_SUPABASE_PUBLISHABLE_KEY'

kallebysantos committed:
stamp: nextjs examples, revert to use cookie options

kallebysantos commented on examples/auth/nextjs/lib/supabase/server.ts:
```diff
@@ -14,9 +14,7 @@ export async function createClient() {
-            cookiesToSet.forEach(({ name, value, options }) =>
-              cookieStore.set(name, value, options)
-            )
+            cookiesToSet.forEach(({ name, value }) => cookieStore.set(name, value))
```
solved cbb697d67140771d2d5d568fad0c3f7fdeaa4918

kallebysantos committed:
fix(docs): typo

kallebysantos committed:
docs: updating nextjs-auth troubleshoot guide to match proxy

ChrisChinchilla committed:
Update apps/docs/content/guides/getting-started/quickstarts/nextjs.mdx

ChrisChinchilla committed:
Revert auth-helpers changes

ChrisChinchilla committed:
Merge branch 'chrischinchilla/docs-555-update-supabase-docs-for-nextjs-16-proxyts-change' of github.com:supabase/supabase into chrischinchilla/docs-555-update-supabase-docs-for-nextjs-16-proxyts-change

ChrisChinchilla committed:
Revert auth-helpers content

ChrisChinchilla committed:
Apply suggestions from code review

ChrisChinchilla committed:
Update apps/docs/content/troubleshooting/how-do-you-troubleshoot-nextjs---supabase-auth-issues-riMCZV.mdx

ChrisChinchilla committed:
Update apps/docs/content/troubleshooting/how-do-you-troubleshoot-nextjs---supabase-auth-issues-riMCZV.mdx

ChrisChinchilla committed:
Update apps/docs/content/troubleshooting/how-do-you-troubleshoot-nextjs---supabase-auth-issues-riMCZV.mdx

ChrisChinchilla committed:
Update apps/docs/content/troubleshooting/how-do-you-troubleshoot-nextjs---supabase-auth-issues-riMCZV.mdx

ChrisChinchilla committed:
Apply suggestions from code review

ChrisChinchilla committed:
Prettier

ChrisChinchilla committed:
Merge branch 'chrischinchilla/docs-555-update-supabase-docs-for-nextjs-16-proxyts-change' of github.com:supabase/supabase into chrischinchilla/docs-555-update-supabase-docs-for-nextjs-16-proxyts-change

dijonmusters commented on apps/docs/content/troubleshooting/how-to-migrate-from-supabase-auth-helpers-to-ssr-package-5NRunM.mdx:
This one needs to be refactored slightly to match the other examples. Again, not something you have caused, but have helped highlight with these changes. The `cookies` function is now async so we need to `await` it when we are creating a Supabase Client serverside. This will affect `lib/supabase/server`, `lib/supabase/proxy` and anywhere we are creating clients from these libs

dijonmusters reviewed (commented):
Looks great! Just one small thing I noticed that we can fix up in a future PR 👍
